subsys/senss/sensor_mgmt.h

```diff
@@ -0,0 +1,194 @@
+/*
+ * Copyright (c) 2022 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef SENSOR_MGMT_H_
+#define SENSOR_MGMT_H_
+
+#include <zephyr/kernel.h>
+#include <zephyr/senss/senss_datatypes.h>
+#include <zephyr/senss/senss_sensor.h>
+#include <zephyr/sys/slist.h>
+
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#define PHANDLE_DEVICE_BY_IDX(idx, node, prop)				\
+	DEVICE_DT_GET(DT_PHANDLE_BY_IDX(node, prop, idx))
+
+#define PHANDLE_DEVICE_LIST(node, prop)					\
+{									\
+	LISTIFY(DT_PROP_LEN_OR(node, prop, 0),				\
+			PHANDLE_DEVICE_BY_IDX,				\
+			(,),						\
+			node,						\
+			prop)						\
+}
+
+#define SENSS_SENSOR_INFO(node)						\
+{									\
+	.dev = DEVICE_DT_GET(node),					\
+	.ord = DT_DEP_ORD(node),					\
+	.info.type = DT_PROP(node, sensor_type),			\
+	.info.sensor_index = DT_PROP_OR(node, sensor_index, 0),		\
+	.info.vendor = DT_NODE_VENDOR_OR(node, NULL),			\
+	.info.model = DT_PROP(node, model),				\
+	.info.name = DT_NODE_FULL_NAME(node),				\
+	.info.friendly_name = DT_PROP(node, friendly_name),		\
+	.info.minimal_interval = DT_PROP(node, minimal_interval),	\
+	.reporter_num = DT_PROP_LEN_OR(node, reporters, 0),		\
+	.reporters = PHANDLE_DEVICE_LIST(node, reporters),		\
+},
+
+#define for_each_sensor(ctx, i, sensor)					\
+	for (i = 0; i < ctx->sensor_num && (sensor = ctx->sensor_db[i]) != NULL; i++)
+
+#define for_each_reporter_conn(i, sensor, conn)			\
+	for (i = 0; i < sensor->conns_num && (conn = &sensor->conns[i]) != NULL; i++)
+
+enum sensor_trigger_mode {
+	SENSOR_TRIGGER_MODE_POLLING = 1,
+	SENSOR_TRIGGER_MODE_DATA_READY = 2,
+};
+
+/**
+ * @struct senss_sensor_dt_info
+ * @brief Sensor device tree data structure
+ */
+struct senss_sensor_dt_info {
+	const struct device *dev;
+	const int ord;
+	struct senss_sensor_info info;
+	uint16_t reporter_num;
+	const struct device *reporters[CONFIG_SENSS_MAX_REPORTER_COUNT];
+};
+
+/**
+ * @struct senss_connection information
+ * @brief senss_connection indicates connection from reporter sensor(source) to client sensor(sink)
+ */
+struct senss_connection {
```

Comment:
is this intended for nanoapps to coalesce sensor data from one real to a virtual? Is the processing function contained in the sink sensor then?

Reply:
yes, correct. All applications who opened device tree defined physical/virtual sensor are also defined as a client.